packages/react/src/primitives/shared/types.ts

```diff
@@ -0,0 +1,174 @@
+  | 'Expander'
+  | 'Flex'
+  | 'Grid'
+  | 'Heading'
+  | 'Icon'
+  | 'Image'
+  | 'Link'
+  | 'Loader'
+  | 'Menu'
+  | 'Pagination'
+  | 'PasswordField'
+  | 'PhoneNumberField'
+  | 'Placeholder'
+  | 'Radio'
+  | 'RadioGroupField'
+  | 'Rating'
+  | 'ScrollView'
+  | 'SearchField'
+  | 'Select'
+  | 'SliderField'
+  | 'SelectField'
+  | 'StepperField'
+  | 'SwitchField'
+  | 'Table'
+  | 'Tabs'
+  | 'Text'
+  | 'TextAreaField'
+  | 'TextField'
+  | 'ToggleButton'
+  | 'ToggleButtonGroup'
+  | 'VisuallyHidden';
+
+export type ComponentClassName =
+  | 'Alert'
+  | 'AlertIcon'
+  | 'AlertHeading'
+  | 'AlertBody'
+  | 'AlertDismiss'
+  | 'Badge'
+  | 'Button'
+  | 'ButtonGroup'
+  | 'ButtonLoaderWrapper'
+  | 'Card'
+  | 'Checkbox'
+  | 'CheckboxButton'
+  | 'CheckboxIcon'
+  | 'CheckboxInput'
+  | 'CheckboxLabel'
+  | 'CheckboxField'
+  | 'Collection'
+  | 'CollectionItems'
+  | 'CollectionSearch'
+  | 'CollectionPagination'
+  | 'CountryCodeSelect'
+  | 'Divider'
+  | 'DividerLabel'
+  | 'Expander'
+  | 'ExpanderContent'
+  | 'ExpanderContentText'
+  | 'ExpanderHeader'
+  | 'ExpanderIcon'
+  | 'ExpanderItem'
+  | 'ExpanderTrigger'
+  | 'Field'
+  | 'FieldDescription'
+  | 'FieldErrorMessage'
+  | 'FieldGroup'
+  | 'FieldGroupControl'
+  | 'FieldGroupOuterEnd'
+  | 'FieldGroupOuterStart'
+  | 'FieldGroupInnerEnd'
+  | 'FieldGroupInnerStart'
+  | 'FieldGroupIcon'
+  | 'FieldGroupIconButton'
+  | 'FieldGroupHasInnerEnd'
+  | 'FieldGroupHasInnerStart'
+  | 'FieldShowPassword'
+  | 'FieldGroupFieldWrapper'
+  | 'Flex'
+  | 'Grid'
+  | 'Heading'
+  | 'Icon'
+  | 'Image'
+  | 'Input'
+  | 'Label'
+  | 'Link'
+  | 'Loader'
+  | 'LoaderDeterminate'
+  | 'LoaderPercentageText'
+  | 'MenuContent'
+  | 'MenuItem'
+  | 'MenuTrigger'
+  | 'Pagination'
+  | 'PaginationItemButton'
+  | 'PaginationItemCurrent'
+  | 'PaginationItemEllipsis'
+  | 'PasswordField'
+  | 'PhoneNumberField'
+  | 'Placeholder'
+  | 'Radio'
+  | 'RadioButton'
+  | 'RadioInput'
+  | 'RadioLabel'
+  | 'RadioGroupField'
+  | 'RadioGroup'
+  | 'Rating'
+  | 'ScrollView'
+  | 'SearchField'
+  | 'SearchFieldClear'
+  | 'SearchFieldSearch'
+  | 'Select'
+  | 'SelectField'
+  | 'SelectWrapper'
+  | 'SelectIconWrapper'
+  | 'SliderField'
+  | 'SliderFieldGroup'
+  | 'SliderFieldLabel'
+  | 'SliderFieldRange'
+  | 'SliderFieldRoot'
+  | 'SliderFieldThumb'
+  | 'SliderFieldTrack'
+  | 'StepperField'
+  | 'StepperFieldButtonDecrease'
+  | 'StepperFieldButtonIncrease'
+  | 'StepperFieldInput'
+  | 'SwitchField'
+  | 'SwitchLabel'
+  | 'SwitchThumb'
+  | 'SwitchTrack'
+  | 'SwitchWrapper'
+  | 'Table'
+  | 'TableCaption'
+  | 'TableBody'
+  | 'TableTd'
+  | 'TableTh'
+  | 'TableFoot'
+  | 'TableHead'
+  | 'TableRow'
+  | 'Tabs'
+  | 'TabItems'
+  | 'Text'
+  | 'Textarea'
+  | 'TextAreaField'
+  | 'TextField'
+  | 'ToggleButton'
+  | 'ToggleButtonGroup'
+  | 'VisuallyHidden';
+
+export type ComponentClasses = string;
+
+interface ComponentClassNameMetadata {
+  className: ComponentClasses;
```

Comment:
Feel like we can just use `string` to reduce the complexity. Is there a particular reason to add one more abstract level here?

Reply:
Looks like the original one is a union of all component classnames, more specific than `string`

Reply:
That's a legitimate call out, will update